mpa/modules/hooks/sam_optimizer_hook.py

```diff
@@ -22,6 +23,12 @@ def __init__(self, rho=0.05, start_epoch=1, **kwargs):
         if rho < 0.0:
             raise ValueError('rho should be greater than 0 for SAM optimizer')
 
+    def before_train_epoch(self, runner):
```

Comment:
Is there any other reason why this should happen in optimizer hook? 
The code written on fp16 and SAM looks the same, but isn't it enough to make it into a separate hook? (create 1 hook and enable?)

Reply:
You saying that it's better to make new hook that functions only insertion of cur_epoch. Is it right?

Reply:
I didn't mean exactly what structure to implement. 
but, what I'm just saying is that the current structure doesn't seem to need to be divided into two hooks.